### PR TITLE
docs: introduces versioning & migration guide

### PR DESCRIPTION
**Pull request by Shipow**
**Summary**

This PR:
- upgrades to docusaurus V2 alpha 61
- refactors custom code for dark theme
- starts fixed version of the docs for 2.6.3. Next is 3.0.0 alpha
- introduces the migration guide
- introduces how to style the DocSeach v3
- makes v2 href import explicit. This point was unclear and we had several supports ticket due to this.

we can start working on the docs for V3 from here.

Dependencies:

- [ ] https://github.com/algolia/docsearch-website/issues/47 once the first stable version is released

### CODE REVIEW

File: docs/migration-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Moving from v2 to v3
+
+This article provides a set of guidelines to migrate from v2 to v3.
+
+### 1. Update the imported JS & CSS libraries
+
+```html
+<!-- Before the closing </head> -->
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css" />
```
Comment (francoischalifour) [inline docs/migration-guide.mdx]: For now, we need the `alpha` tag.

```suggestion
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@docsearch/css@alpha" />
```

File: docs/migration-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+<!-- Before the closing </body> -->
+<script src="https://cdn.jsdelivr.net/npm/@docsearch/js"></script>
```
Comment (francoischalifour) [inline docs/migration-guide.mdx]:
```suggestion
<script src="https://cdn.jsdelivr.net/npm/@docsearch/js@alpha"></script>
```

File: docs/migration-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+<script>
+  docsearch({
+    // See next section for upgrading this JS snippet
+  });
+</script>
+```
+
+### 2. Update the JS snippet to intiate DocSearch
```
Comment (francoischalifour) [inline docs/migration-guide.mdx]:
```suggestion
### 2. Update the JS snippet to instantiate DocSearch
```

File: docs/migration-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+#### Replace `inputSelector` with `container`
+
+The `container` argument is a [CSS selector][1] matching the HTML element to
+feature. This element used to be an `<input/>` with DocSearch v2. The JS snippet
+call replaces the first matching element with the DocSearch button. This button
+opens the search modal.
+
+For example, the minimalist required structure and imports is:
```
Comment (francoischalifour) [inline docs/migration-guide.mdx]:
```suggestion
For example, the minimalist required structure and imports are:
```

File: docs/modal.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
+title: Modal Search-UI
```
Comment (francoischalifour) [inline docs/modal.md]: Why have a `-` in between the two words?

```suggestion
title: Modal Search UI
```

File: docs/modal.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+Once your Algolia DocSearch index is ready, set up, and filled with the right
+data, you will need to integrate our dedicated Search-UI. To add this modal
```
Comment (francoischalifour) [inline docs/modal.md]:
```suggestion
data, you will need to integrate our dedicated search UI. To add this modal
```

File: docs/modal.md (hunk 5 above)
Comment (francoischalifour) [inline docs/modal.md]: Should we rather call this page "search-ui.md" and "Search UI"?

File: docs/run-your-own.md [hunk 7]
```diff
@@ -135,8 +135,8 @@ the same result by running a live server in the `./playground` subdirectory.\`_
 
 ## Integration
 
-Once you're satisfied with your config, you can integrate the dropdown menu in
-your website by following the [instructions here][14].
+Once you're satisfied with your config, you can integrate the modal search in
```
Comment (francoischalifour) [inline docs/run-your-own.md]: There's no need to refer to it as the "modal search", it's a "search".

```suggestion
Once you're satisfied with your config, you can integrate the search in
```

File: docs/styling.mdx [hunk 8]
```diff
@@ -4,102 +4,89 @@ title: Styling DocSearch
 
 import useBaseUrl from '@docusaurus/useBaseUrl';
 
-DocSearch default color scheme comes in a grey theme with blue highlight.
+DocSearch v3 comes with 2 default color scheme. One for the light mode and one
+for the dark mode.
```
Comment (francoischalifour) [inline docs/styling.mdx]:
```suggestion
DocSearch v3 comes with two default color themes: light mode and dark mode.
```

File: docs/migration-guide.mdx (hunk 1 above)
Comment (s-pace) [reply docs/migration-guide.mdx]: Reflected in the dependency #47

**Comment by shortcuts** [comment]
Closing since the documentation has moved to https://github.com/algolia/docsearch.

**Pull request closed without merging**